FEM/ShapeFunctionPool.h

```diff
@@ -0,0 +1,68 @@
+/*! \file ShapeFunctionPool.h
+    \brief Compute shape functions and their gradients with respect to
+     the local coodinates, and store the results
+
+     \author Wenqing Wang
+     \date Feb. 2015
+ 
+     \copyright
+      Copyright (c) 2015, OpenGeoSys Community (http://www.opengeosys.org)
+             Distributed under a Modified BSD License.
+             See accompanying file LICENSE.txt or
+             http://www.opengeosys.org/project/license
+*/
+#ifndef OGS_SHAPEFUNCTIONPOOL_H
+#define OGS_SHAPEFUNCTIONPOOL_H
+
+#include <vector>
+
+#include "MSHEnums.h"
+
+namespace FiniteElement
+{
+class CElement;
+
+class ShapeFunctionPool
+{
+public:
+	/*!
+		\param elem_types          All involved element types.
+		\param quadrature          Numerical integration object.
+		\param num_sample_gs_pnts  Number of sample Gauss points.
+	*/
+	ShapeFunctionPool(const std::vector<MshElemType::type>& elem_types,
+		              CElement& quadrature, const int num_sample_gs_pnts);
```

Comment:
The name of CElement is not proper, the class actually provides the computation of shape functions and their derivatives etc. Therefore I the second argument is named as ` quadrature`.